src/crypto/index.js

```diff
@@ -1018,26 +1030,26 @@ Crypto.prototype.backupGroupSession = function(
         throw new Error("Key backups are not enabled");
     }
 
-    const data = this._backupPayloadForSession(
-        senderKey, forwardingCurve25519KeyChain,
-        sessionId, sessionKey, keysClaimed,
-        exportFormat,
-    );
-    return this._baseApis.sendKeyBackup(roomId, sessionId, this.backupInfo.version, data);
+    await this._cryptoStore.markSessionsNeedingBackup([{
+        senderKey: senderKey,
+        sessionId: sessionId,
+    }]);
+
+    this._maybeSendKeyBackup();
 };
 
 Crypto.prototype.backupAllGroupSessions = async function(version) {
-    const keys = await this.exportRoomKeys();
-    const data = {};
-    for (const key of keys) {
-        if (data[key.room_id] === undefined) data[key.room_id] = {sessions: {}};
-
-        data[key.room_id]['sessions'][key.session_id] = this._backupPayloadForSession(
-            key.sender_key, key.forwarding_curve25519_key_chain,
-            key.session_id, key.session_key, key.sender_claimed_keys, true,
-        );
-    }
-    return this._baseApis.sendKeyBackup(undefined, undefined, version, {rooms: data});
+    await this._cryptoStore.doTxn(
+        'readwrite', [IndexedDBCryptoStore.STORE_SESSIONS, IndexedDBCryptoStore.STORE_BACKUP], (txn) => {
+            this._cryptoStore.getAllEndToEndInboundGroupSessions(txn, (session) => {
+                if (session !== null) {
+                    this._cryptoStore.markSessionsNeedingBackup([session], txn);
+                }
+            });
+        }
+    );
+
+    this._maybeSendKeyBackup();
```

Comment:
should await on this so the `backupAllGroupSessions` promise doesn't complete until the backup is done